liquibase/src/test/groovy/io/micronaut/liquibase/LiquibaseMigrationRunnerSpec.groovy

```diff
@@ -0,0 +1,48 @@
+package io.micronaut.liquibase
+
+import io.micronaut.context.annotation.Primary
+import io.micronaut.context.annotation.Requires
+import io.micronaut.jdbc.DataSourceResolver
+import javax.inject.Singleton
+import javax.sql.DataSource
+
+class LiquibaseMigrationRunnerSpec extends ApplicationContextSpecification {
+
+    @Override
+    String getSpecName() {
+        'LiquibaseMigrationRunnerSpec'
+    }
+    @Override
+    Map<String, Object> getConfiguration() {
+        super.configuration +
+                getDataSourceConfiguration('liquibaseDisabledDb') +
+                getJpaConfiguration(['example.micronaut'])
+    }
+    void "LiquibaseMigrationRunner::onCreated returns wrapped Datasource"() {
+        when:
+        DataSource dataSource = applicationContext.getBean(DataSource)
+
+        then:
+        !(dataSource instanceof ReturnedByDataSourceResolver)
+    }
+
+    @Primary
+    @Requires(property = 'spec.name', value = 'LiquibaseMigrationRunnerSpec')
+    @Singleton
+    static class MockDataSourceResolver implements DataSourceResolver {
+        @Override
+        DataSource resolve(DataSource dataSource) {
+            new ReturnedByDataSourceResolver(dataSource)
+        }
+    }
+
+    static class ReturnedByDataSourceResolver implements DataSource {
+
+        @Delegate
+        private final DataSource dataSource
+
+        ReturnedByDataSourceResolver(DataSource dataSource) {
+            this.dataSource = dataSource
+        }
+    }
+}
```

Comment:
Why mocking the `DataSourceResolver` instead of using the real one?

Reply:
It was the easiest way to test. At least, what I a came up. 

The test verifies that `LiquibaseMigrationRunner::onCreated` does not return of `DataSourceResolver::resolve`instead it returns `event::getBean()`